gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStoreFactory.java

```diff
@@ -38,6 +41,47 @@
   protected static final String SIMPLE_HDFS_SCHEME_PREFIX = "simple-";
   protected static final String HDFS_SCHEME_NAME = "hdfs";
 
+  public final static String DEFAULT_STORE_URI_KEY = "default_store_uri";
+
+  public SimpleHDFSConfigStoreFactory() {
+    this(ConfigFactory.empty());
+  }
+
+  private final Optional<URI> defaultStoreURI;
+  private final FileSystem defaultStoreFS;
+
+  public SimpleHDFSConfigStoreFactory(Config factoryConfig) {
+    try {
+      if (factoryConfig.hasPath(DEFAULT_STORE_URI_KEY)) {
+        String uri = factoryConfig.getString(DEFAULT_STORE_URI_KEY);
+        Path defStorePath = new Path(uri);
```

Comment:
Is it possible for DEFAULT_STORE_URI_KEY to be set to null and we get a NPE here?
You can use `ConfigUtils.hasNonEmptyPath(factoryConfig, DEFAULT_STORE_URI_KEY)` instead of `factoryConfig.hasPath(DEFAULT_STORE_URI_KEY)`